api/v1alpha1/hosted_controlplane.go

```diff
@@ -95,6 +95,11 @@ type HostedControlPlaneSpec struct {
 	// cluster when applicable.
 	// +optional
 	SecretEncryption *SecretEncryptionSpec `json:"secretEncryption,omitempty"`
+
+	// ExposeKubeadminPassword specifies whether to expose the kubeadmin password
+	// for the guest cluster.
+	// +optional
+	ExposeKubeadminPassword bool `json:"exposeKubeadminPassword,omitempty"`
```

Comment:
golang `bool` defaults to `false` so this is a breaking change, as we will now by default not expose this anymore. IMHO, the default should remain unchanged. You have two options to achieve that:
* Rename the option so that it represents the inverse of what is currently suggested
* Make it a bool pointer, this allows to differentiate `unset` and `false`, then add some code to default

I'd prefer the second option, as it allows to to default to either rather than implicitly hardcoding one defaulting